src/librustc/ty/layout.rs

```diff
@@ -906,6 +911,16 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
                     return Ok(tcx.intern_layout(st));
                 }
 
+                let mut align = dl.i8_align;
+                let mut max_size = Size::ZERO;
+
+                for fields in variants.iter() {
+                    let (v_size, v_align) = self.variant_size(fields, &def.repr)
+                        .ok_or(LayoutError::SizeOverflow(ty))?;
```

Comment:
This looks like it could be expensive.

Reply:
It also doesn't take field sorting and the offset for the tag into account?

Reply:
It considers field sorting, but it does not compute it. If sorting happens, all padding can be at the start.

This also means it's relatively cheap.